testflinger/__init__.py

```diff
@@ -153,4 +138,47 @@ def unhandled_exception(exc):
     return tf_app
 
 
+def setup_mongodb(application):
+    """
+    Setup mongodb connection if we have valid config data
+    Otherwise leave it empty, which means we are probably running unit tests
+    """
+    mongo_user = os.environ.get("MONGO_USER")
+    mongo_pass = os.environ.get("MONGO_PASSWORD")
+    mongo_db = os.environ.get("MONGO_DATABASE")
+    if not application.config.get("MONGO_URI") and not (
+        mongo_user and mongo_pass and mongo_db
+    ):
+        # We are probably running unit tests
+        return
+
+    if not application.config.get("MONGO_URI"):
+        application.config[
+            "MONGO_URI"
+        ] = f"mongodb://{mongo_user}:{mongo_pass}@mongo:27017/{mongo_db}"
+
+    mongo_client = PyMongo(
+        application,
+        uri=application.config["MONGO_URI"],
+        uuidRepresentation="standard",
+    )
```

Comment:
We haven't needed to do this so far, because the connections are just between instances on the backend where the thing we're deploying on (whether it be docker/k8s/openstack/whatever should control access to only allow connections from testflinger->mongodb, so there shouldn't be a need to mitigate risk for something sitting in the middle of that. For the connections to the API server itself, we definitely should (and do) use ssl now. Is there something I'm not considering here for why we should consider adding this anyway?  If it doesn't become something that just complicates the configuration for a feature we don't use, then I'm happy to add it.

Reply:
Sorry if this request is ignorant. I’m not aware of the topology of the network, where the database server and the API server are located. If they are located on the same host, we don’t need SSL/TLS. If they are located on the separate hosts, I just suggest to use TLS for communications with DB according to the Zero Trust Architecture. However, it’s up to you

Reply:
It's not ignorant at all! and I agree it's a good idea. I just wonder if it's a priority on an app that is completely internal and has no auth mechanism - thus you could get at the same data by going through the front door rather than by getting in the middle of a transaction between two backend services that are restricted to only talking to each other.  Just because I don't think it's a high priority today though, doesn't mean it won't ever be. I think we should add it to the backlog as something to consider in the future.

Reply:
Okay, it's a good idea to add authentication mechanisms in the future.

Reply:
I've added a card for this to the backlog